langchain_tools_demo/main.py

```diff
@@ -95,6 +93,20 @@ async def chat_handler(request: Request, prompt: str = Body(embed=True)):
         raise HTTPException(status_code=500, detail=f"Error invoking agent: {err}")
 
 
+@app.post("/reset")
+async def reset(request: Request):
+    """Reset agent"""
+    global user_agents
+    uuid = request.session["uuid"]
+    if uuid in user_agents.keys():
```

Comment:
if uuid isn't in user_agents, should we throw an error?

Reply:
That way you would get an error when you click the reset button first before start chatting with the agent (the agent hasn't been initialized yet). This may not be desired.

Reply:
I thought the agent was initialized when we load the index ("/") -- doesn't that mean the agent should always been initalized? Are we redirecting them to the index after we call this endpoint?

Reply:
Currently the agent is initialized in the "/chat" handler on line 83. We could move this part to the index (I think it would be more intuitive)

Reply:
Ah GitHub shrank the sections for me so I missed the function declaration. 

I agree -- let's move it to index.

Reply:
moved to index and raised exception in "/reset" if agent not found.